File: leetcode/FindKDistantIndices.java
class Solution {
    public List<Integer> findKDistantIndices(int[] nums, int key, int k) {

        List<Integer> result = new ArrayList<>();
        
        List<Integer> equalKey = new ArrayList<>();
        
        for (int i = 0; i <nums.length; i++){
            if (nums[i] == key){
                equalKey.add(i);
            }
        }
        
        
        for (int i = 0; i < nums.length; i++){
            for (int elem : equalKey){
                if (Math.abs(i - elem) <= k){
                    result.add(i);
                    break;
                }
            }
            
        }
        
        Collections.sort(result);
        
        return result;
    }
}

File: leetcode/findKDistantIndices.py
class Solution:
    def findKDistantIndices(self, nums: List[int], key: int, k: int) -> List[int]:
        
        indicies = [] 
        
        for x in range(0, len(nums)):
            if nums[x] == key:
                indicies.append(x)
                
        output = [ ]
        
        for x in range(0, len(nums)):
            
            for index in indicies: 
                
                is_index = False
                
                if abs(x - index) <= k:
                    output.append(x)
                    break
                    
        return sorted(output)